src/modules/auth/controllers/auth.controller.ts

```diff
@@ -123,6 +129,94 @@ class AuthController extends BaseController {
       },
     );
   }
+
+  private async forgetPassword() {
+    this.router.post(
+      "/v1/auth/forget-password",
+      async (req: Request, res: Response) => {
+        const reqBody = this.validate(req, res, ForgetPasswordSchema);
+        if (!reqBody) {
+          return;
+        }
+
+        const user = await this.userSubservice.findOne({
+          email: reqBody.email,
+        });
+        if (!user) {
+          return this.handleError(res, {
+            statusCode: StatusNotFound,
+            message: "User not found",
+          });
+        }
+
+        const resetToken = uuidv4();
+
+        await this.redisClient.set(
+          `forget-password:${resetToken}`,
+          user._id as string,
+          "EX",
+          300,
+        );
+
+        // TODO: Send password reset email to the user with the reset token (e.g., using SendGrid, Mailgun)
+
+        return this.handleSuccess(res, {
+          message: "Password reset email sent",
+        });
+      },
+    );
+  }
+
+  private async resetPassword() {
+    this.router.post(
+      "/v1/auth/reset-password",
+      async (req: Request, res: Response) => {
+        const reqBody = this.validate(req, res, ResetPasswordSchema);
+        if (!reqBody) {
+          return;
+        }
+
+        const { resetToken, newPassword } = reqBody;
+
+        const userId = await this.redisClient.get(
+          `forget-password:${resetToken}`,
+        );
+
+        if (!userId) {
+          return this.handleError(res, {
+            statusCode: StatusNotFound,
+            message: "Invalid or expired reset token",
+          });
+        }
+
+        const user = await this.userSubservice.findOne({ _id: userId });
+
+        if (!user) {
+          return this.handleError(res, {
+            statusCode: StatusNotFound,
+            message: "User not found",
+          });
+        }
+
+        const hashedPassword = await bcrypt.hash(newPassword as string, 10);
+
+        const updatedUser = await this.userSubservice.update(
+          { _id: userId },
+          { password: hashedPassword },
+        );
+
+        if (this.isServiceError(res, updatedUser)) {
+          return;
+        }
+
+        await this.redisClient.del(`forget-password:${resetToken}`);
```

Comment:
```suggestion
        this.redisClient.del(`forget-password:${resetToken}`);
```
doesn't need to be await